cypress/e2e/downloads.cy.ts

```diff
@@ -0,0 +1,369 @@
+        expect(requestCount).to.equal(0);
+      });
+    });
+    describe('entry id', () => {
+      it('should use current entry id', async () => {
+        await loadPlayerAndSetMedia();
+        expect(requestUrl.indexOf('entryId/1234')).to.not.equal(-1);
+      });
+    });
+    describe('ks', () => {
+      it('should contain ks from config if ks is set', async () => {
+        await loadPlayerAndSetMedia();
+        expect(requestUrl.indexOf(`ks/5678`)).to.not.equal(-1);
+      });
+      it('should not make download call if ks is not set', async () => {
+        await loadPlayerAndSetMedia(null, {ks: ''});
+        expect(requestCount).to.equal(0);
+      });
+    });
+    describe('flavor id', () => {
+      it('should use flavor id if it is set', async () => {
+        await loadPlayerAndSetMedia({plugins: {uiManagers: {}, download: {flavorId: '1234'}}});
+        expect(requestUrl.indexOf('flavorId')).to.not.equal(-1);
+      });
+      it('should not contain a flavor id if flavor id is not set', async () => {
+        await loadPlayerAndSetMedia();
+        expect(requestUrl.indexOf('flavorId')).to.equal(-1);
+      });
+    });
+    describe('flavor param id', () => {
+      it('should use flavor param id if it is set', async () => {
+        await loadPlayerAndSetMedia({plugins: {uiManagers: {}, download: {flavorParamId: '1234'}}});
+        expect(requestUrl.indexOf('flavorParamId/1234')).not.to.equal(-1);
+      });
+      it('should use default flavor param id if it is not set', async () => {
+        await loadPlayerAndSetMedia();
+        expect(requestUrl.indexOf('flavorParamId/0')).not.to.equal(-1);
+      });
+    });
+  });
+
+  describe('download overlay button', () => {
+    describe('if content type is not video', () => {
+      beforeEach(() => {
+        cy.intercept('**/playManifest/**', {
+          headers: {
+            'content-type': 'image/gif'
+          },
+          fixture: 'video.mp4'
+        });
+      });
+
+      it('should be hidden', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay-button"]').should('not.exist');
+        });
+      });
+    });
+
+    describe('if content type is video', () => {
+      beforeEach(() => {
+        cy.intercept('**/playManifest/**', {fixture: 'video.mp4'});
+      });
+
+      it('should be visible', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay-button"]').should('exist');
+        });
+      });
+
+      it('should show overlay on click', () => {
+        return loadPlayerAndSetMedia().then(() => {
+          cy.get('[data-testid="download-overlay"]').should('not.exist');
+          cy.get('[data-testid="download-overlay-button"]').click({force: true});
+          cy.get('[data-testid="download-overlay"]').should('exist');
+        });
+      });
+    });
+
+    it('should show overlay on click', () => {
```

Comment:
@SivanA-Kaltura  is this the same test as the one above it?

Reply:
yes, removing it